tests/resspect/test_query_strategies.py

```diff
@@ -0,0 +1,152 @@
+import pytest
+
+from resspect.query_strategies import (
+    random_sampling,
+    uncertainty_sampling,
+    uncertainty_sampling_entropy,
+    uncertainty_sampling_least_confident,
+    uncertainty_sampling_margin,
+)
+
+
+@pytest.mark.parametrize("batch_size,queryable",list(itertools.product([1, 5, 10], [True, False])))
+def test_random_sampling(batch_size, queryable):
+    """Test the random sampling functionality."""
+    test_ids = np.arange(0, 100)
+    queryable_ids = test_ids[test_ids % 3 == 0]
+
+    # Test that we generate the correct number of samples.
+    sample = random_sampling(test_ids, queryable_ids, batch=batch_size, queryable=queryable)
+    assert len(sample) == batch_size
+    assert len(np.unique(sample)) == batch_size
+
+    if queryable:
+        assert np.all(np.array(sample) % 3 == 0)
+
+
+def test_uncertainty_sampling():
+    """Test the uncertainity sampling functionality."""
+    test_ids = np.arange(0, 10)
+    queryable_ids = np.array([0, 1, 2, 3, 4, 7, 8, 9])  # No 5 or 6
+    class1_probs = np.array([
+        0.01,  # 0 - very low
+        0.50,  # 1 - very high
+        0.10,  # 2 - low
+        0.20,  # 3 - low
+        0.65,  # 4 - medium high
+        0.45,  # 5 - very high
+        0.25,  # 6 - medium
+        0.80,  # 7 - low
+        0.40,  # 8 - high
+        0.02,  # 9 - very low
+    ])
+    class_probs = np.array([class1_probs, 1.0 - class1_probs]).T
+
+    # Test that we generate the correct number of samples.
+    sample = uncertainty_sampling(class_probs, test_ids, queryable_ids, batch=3)
+    assert len(sample) == 3
+    assert np.array_equal(sample, [1, 8, 4])
+
+
+@pytest.mark.parametrize("batch_size",[1, 5, 10, 20])
+def test_uncertainty_sampling_entropy_random(batch_size):
+    """Test the entropy-based uncertainty sampling functionality with random data."""
+    num_samples = 100
+    num_classes = 5
+    test_ids = np.arange(0, 100)
+    queryable_ids = test_ids[test_ids % 3 == 0]
+
+    # Generate class probabilities.
+    np.random.seed(100)
+    class_prob = np.random.random((num_samples, num_classes))
+    normalized_probs =  class_prob / np.tile(np.sum(class_prob, axis=1), (num_classes, 1)).T
+
+    # Test that we generate the correct number of samples.
+    sample = uncertainty_sampling_entropy(
+        normalized_probs,
+        test_ids,
+        queryable_ids,
+        batch=batch_size
+    )
+    assert len(sample) == batch_size
+    assert len(np.unique(sample)) == batch_size
+    assert np.all(np.array(sample) % 3 == 0)
+
+
+def test_uncertainty_sampling_entropy_known():
+    """Test the entropy-based uncertainty sampling functionality with known entropies."""
+    test_ids = np.arange(0, 8)
+    queryable_ids = np.arange(0, 8)
+    class_prob = np.array(
+        [
+            [1.0, 0.0, 0.0],              # 0.0
+            [0.5, 0.5, 0.0],              # 0.693
+            [1.0/3.0, 1.0/3.0, 1.0/3.0],  # 1.098
+            [0.5, 0.0, 5.0],              # 0.693
```

Comment:
Not totally sure what this function is doing, but should this line be  `[0.5, 0.0, 0.5]` instead of`[0.5, 0.0, 5.0]`?

Reply:
Okay, I think I understand now. Shouldn't change the results, but would be good to fix

Reply:
Good catch. I correct this.